src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Public/INLHandlerFactory.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System.Collections.Specialized;
+using Microsoft.PowerFx.Intellisense;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Public
+{
+    public interface INLHandlerFactory
+    {
+        public NLHandler GetNlHandler(IPowerFxScope uri);
```

Comment:
>IPowerFxScope [](http://example.com/codeflow?start=38&length=13)

this makes sense. 
Scope uri tells us which formula bar we have, and the context could be dependent on that.